flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -185,16 +186,19 @@ function set_conf_ssl {
     echo "Using SAN ${SANSTRING}"
 
     # create certificates
-    keytool -genkeypair -alias ca -keystore "${TEST_DATA_DIR}/ssl/ca.keystore" -dname "CN=Sample CA" -storepass password -keypass password -keyalg RSA -ext bc=ca:true
+    keytool -genkeypair -alias ca -keystore "${TEST_DATA_DIR}/ssl/ca.keystore" -dname "CN=Sample CA" -storepass password -keypass password -keyalg RSA -ext bc=ca:true -storetype PKCS12
     keytool -keystore "${TEST_DATA_DIR}/ssl/ca.keystore" -storepass password -alias ca -exportcert > "${TEST_DATA_DIR}/ssl/ca.cer"
     keytool -importcert -keystore "${TEST_DATA_DIR}/ssl/ca.truststore" -alias ca -storepass password -noprompt -file "${TEST_DATA_DIR}/ssl/ca.cer"
 
-    keytool -genkeypair -alias node -keystore "${TEST_DATA_DIR}/ssl/node.keystore" -dname "CN=${NODENAME}" -ext SAN=${SANSTRING} -storepass password -keypass password -keyalg RSA
+    keytool -genkeypair -alias node -keystore "${TEST_DATA_DIR}/ssl/node.keystore" -dname "CN=${NODENAME}" -ext SAN=${SANSTRING} -storepass password -keypass password -keyalg RSA -storetype PKCS12
     keytool -certreq -keystore "${TEST_DATA_DIR}/ssl/node.keystore" -storepass password -alias node -file "${TEST_DATA_DIR}/ssl/node.csr"
     keytool -gencert -keystore "${TEST_DATA_DIR}/ssl/ca.keystore" -storepass password -alias ca -ext SAN=${SANSTRING} -infile "${TEST_DATA_DIR}/ssl/node.csr" -outfile "${TEST_DATA_DIR}/ssl/node.cer"
     keytool -importcert -keystore "${TEST_DATA_DIR}/ssl/node.keystore" -storepass password -file "${TEST_DATA_DIR}/ssl/ca.cer" -alias ca -noprompt
     keytool -importcert -keystore "${TEST_DATA_DIR}/ssl/node.keystore" -storepass password -file "${TEST_DATA_DIR}/ssl/node.cer" -alias node -noprompt
 
+    # keystore is converted into a pem format to use it as node.pem with curl in Flink REST API queries, see also $CURL_SSL_ARGS
+    openssl pkcs12 -passin pass:password -in "${TEST_DATA_DIR}/ssl/node.keystore" -out "${TEST_DATA_DIR}/ssl/node.pem" -nodes
```

Comment:
`start_cluster` uses `query_running_tms` in `common.sh`.
`query_running_tms` has to use certs in `curl` command if SSL is enabled (previously it was not a problem because if `"x$USE_SSL" = "xON"` was wrong and used http url but failed curl was ignored in broken timeout'ed waiting loop, I think along with cert exceptions in JM logs).
`openssl` generates `node.pem` used with `curl` in `CURL_SSL_ARGS` to properly connect to SSL'ed cluster.
`keytool` uses by default deprecated `JKS`, this fact was also logged previously in test run.
From what I found for 1.7, `openssl` uses another changed non-deprecated protocol to generate understandable `node.pem` for `curl`.

Reply:
Alright, this makes sense. Thanks for the clarification @azagrebin.